redskyctl/internal/commands/generate/experiment/containerresources.go

```diff
@@ -221,3 +221,21 @@ func materializeResourceList(node *yaml.MapNode) corev1.ResourceList {
 	}
 	return resources.Requests
 }
+
+// toIntWithRange returns the quantity as an int value with a default range.
+func toIntWithRange(name corev1.ResourceName, q resource.Quantity) (value *intstr.IntOrString, min int32, max int32) {
+	var scaled intstr.IntOrString
+
+	switch name {
+	case corev1.ResourceMemory:
+		scaled = intstr.FromInt(int(q.ScaledValue(resource.Mega)))
+		min = int32(math.Pow(2, math.Floor(math.Log2(float64(scaled.IntVal/2)))))
```

Comment:
This feels clever. 
Am I understanding it right?
min = 2^(val/2 taking care to end on a bit boundary)
max = 2^(val*2 taking care to end on a bit boundary)

Reply:
Yeah, just sticking with powers of 2 for memory values but not CPU. Which reminds me: I did not account for replicas.

Reply:
LGTM, my main issue here is that we don't enforce limits in the genereated min max. for example if i set a baseline of 1050 i get max= pow(2, ceil(log2(1050*2)))=4096. Do i really want to have 4GB of memory requested?